### PR TITLE
propose HIPE: DIDComm explainer

### PR DESCRIPTION
**Pull request by dhh1128**

### CODE REVIEW

**Comment by TelegramSam** [comment]
I feel like this is ready for FCP. It doesn't answer all questions but is a fine foundation that can be amended through future HIPEs and PRs.

File: text/0003-did-comm/README.md [hunk 1]
```diff
@@ -0,0 +1,215 @@
...
+DIDComm--connecting and maintaining relationships, issuing credentials,
+providing proof, etc.--are called __protocols__; they are described [elsewhere](
+https://github.com/hyperledger/indy-hipe/pull/69).
+
+#### Rough Overview
+
+A typical DIDComm interaction works like this:
+
+<blockquote>
+Imagine Alice wants to negotiate with Bob to sell something online, and
```
Comment (swcurran) [inline text/0003-did-comm/README.md]: I suggest that in this example, we use a protocol that is in process. Perhaps the tic-tac-toe one.  That way at the end we can point to it for details of how a protocol gets specified.

File: text/0003-did-comm/README.md [hunk 2]
```diff
@@ -0,0 +1,215 @@
...
+
+That's it.
+
+Well, mostly. The description is pretty good, if you squint, but it does
+not fit all DIDComm interactions:
+
+* DIDComm doesn't always involve turn-taking and request-response.
+* DIDComm interactions can involve more than 2 parties, and the parties are
+not always individuals.
+* DIDComm may include formats other than JSON.
```
Comment (swcurran) [inline text/0003-did-comm/README.md]: I'm not sure why this is needed at this point (it seems several levels of detail below this overview). Further, even if the data format is not JSON, it's delivered embedded in JSON.  As such, I don't think as stated, this is correct.

**Comment by dhh1128** [comment]
This PR has been superseded by Aries RFC 0005 (https://github.com/hyperledger/aries-rfcs/tree/master/concepts/0005-didcomm)

**Pull request closed without merging**